src/app/components/itinerary-form/itinerary-form.component.html

```diff
@@ -0,0 +1,140 @@
+<div class="card card-compact mb-2 w-full bg-gray-300 shadow-xl md:w-96">
+  <div class="card-body">
+    <h2 class="card-title">Create an Activity!</h2>
+    <button
+      type="button"
+      class="absolute right-4 top-4 flex items-center gap-1 text-blue-700"
+      onclick="tagModal.showModal()">
+      @if (activityForm.value.tag && activityForm.value.tag !== '') {
+        {{ activityForm.value.tag }}
+        <mat-icon fontIcon="edit"></mat-icon>
+      } @else {
+        Add tag
+        <mat-icon fontIcon="add_circle"></mat-icon>
+      }
+    </button>
+    <div [formGroup]="activityForm">
+      <dialog id="tagModal" class="modal">
+        <div class="modal-box">
+          <form method="dialog">
+            <button
+              type="button"
+              (click)="cancelTag()"
+              class="btn btn-circle btn-ghost btn-sm absolute right-4 top-4">
+              ✕
+            </button>
+          </form>
+          <h3 class="text-lg font-bold">Create Tag!</h3>
+          <mat-form-field hideRequiredMarker="true" class="mt-2 w-full">
+            <mat-label>Tag:</mat-label>
+            <input
+              type="text"
+              matInput
+              placeholder="Type Here..."
+              formControlName="tag" />
+            <mat-error>Please enter a tag</mat-error>
+          </mat-form-field>
+          <p class="py-4">Tap on ✕ to cancel.</p>
+          <form method="dialog" class="flex justify-end">
+            <button
+              type="button"
+              class="btn btn-success btn-sm"
+              onclick="tagModal.close()">
```

Comment:
Not very angular of you. `onclick` is plain jane html/js

Reply:
I can do it using (click) yes, but this is how daisy did it so I left it as is:
https://daisyui.com/components/modal/

Reply:
Please angularify it rather